internal/pkg/transformation/kubernetes_label_filter_test.go

```diff
@@ -0,0 +1,427 @@
+				"tier":                      "frontend",
+				"app.kubernetes.io/name":    "my-app",
+				"app.kubernetes.io/version": "1.0.0",
+				"build-id":                  "abc123",
+			},
+			expectedLabels: map[string]string{
+				"app_kubernetes_io_name":    "my-app",
+				"app_kubernetes_io_version": "1.0.0",
+			},
+		},
+		{
+			name:              "FilterMultiplePatterns",
+			allowlistPatterns: []string{"^app$", "^tier$", "^environment$"},
+			podLabels: map[string]string{
+				"app":               "nginx",
+				"tier":              "frontend",
+				"environment":       "prod",
+				"build-id":          "abc123",
+				"pod-template-hash": "xyz789",
+			},
+			expectedLabels: map[string]string{
+				"app":         "nginx",
+				"tier":        "frontend",
+				"environment": "prod",
+			},
+		},
+		{
+			name:              "FilterCommonK8sLabels",
+			allowlistPatterns: []string{"^app\\.kubernetes\\.io/.*", "^version$"},
+			podLabels: map[string]string{
+				"app.kubernetes.io/name":      "my-app",
+				"app.kubernetes.io/component": "backend",
+				"app.kubernetes.io/version":   "2.0.0",
+				"version":                     "v2",
+				"build-id":                    "xyz",
+				"controller-revision-hash":    "abc",
+			},
+			expectedLabels: map[string]string{
+				"app_kubernetes_io_name":      "my-app",
+				"app_kubernetes_io_component": "backend",
+				"app_kubernetes_io_version":   "2.0.0",
+				"version":                     "v2",
+			},
+		},
+		{
+			name:              "NoMatchingLabels",
+			allowlistPatterns: []string{"^nonexistent$"},
+			podLabels: map[string]string{
+				"app":  "nginx",
+				"tier": "frontend",
+			},
+			expectedLabels: map[string]string{},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a fake pod with the test labels
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: "default",
+					UID:       types.UID("test-uid-123"),
+					Labels:    tt.podLabels,
+				},
+			}
+
+			// Create fake Kubernetes client
+			fakeClient := fake.NewSimpleClientset(pod)
+
+			// Create PodMapper with allowlist config
+			config := &appconfig.Config{
+				KubernetesEnablePodLabels:        true,
+				KubernetesPodLabelAllowlistRegex: tt.allowlistPatterns,
+			}
+
+			podMapper := &PodMapper{
+				Config:           config,
+				Client:           fakeClient,
+				labelFilterCache: newLabelFilterCache(config.KubernetesPodLabelAllowlistRegex),
+			}
+
+			// Call getPodMetadata
+			metadata, err := podMapper.getPodMetadata("default", "test-pod")
+			require.NoError(t, err, "getPodMetadata should not return error")
+			require.NotNil(t, metadata, "metadata should not be nil")
+
+			// Verify UID is set correctly
+			assert.Equal(t, "test-uid-123", metadata.UID, "UID should match")
+
+			// Verify labels match expected (after filtering and sanitization)
+			assert.Equal(t, len(tt.expectedLabels), len(metadata.Labels),
```

Comment:
```suggestion
			assert.Equal(t, tt.expectedLabels, metadata.Labels)
```

Do we need all of these assertion loops if we can just assert that the map we expect is the map we get? That should make for a more robust assertion

Reply:
Good call, cleaned up these assertions.